pkg/backup/backup.go

```diff
@@ -154,6 +165,13 @@ type backupContext struct {
 	// resource, from either the networking.k8s.io or extensions api groups. We only want to back them
 	// up once, from whichever api group we see first.
 	networkPoliciesBackedUp bool
+
+	// backedUpItems keeps track of items that have been backed up already. The string should be derived
+	// by calling String() on an itemKey instance representing the item.
+	backedUpItems sets.String
+
+	// additionalItems is a list of items returned by Actions that should be additionally backed up.
+	additionalItems []itemKey
```

Comment:
Recording here for posterity - might want to consider refactoring the backup code to use a queue + workers, so backing up anything, whether as part of the normal process or "additional" one, would go through a single queue.